src/Brahma.FSharp.OpenCL.Core/Kernel.fs

```diff
@@ -0,0 +1,158 @@
+namespace Brahma.FSharp.OpenCL
+
+open OpenCL.Net
+
+open Microsoft.FSharp.Quotations
+open FSharp.Quotations.Evaluator
+open Brahma.FSharp.OpenCL.Translator
+
+type GpuKernel<'TRange, 'a, 't when 'TRange :> INDRangeDimension>
```

Comment:
As far as I understand, type parameter `'t` needs to avoid rewriting translator and allows accepting `Buffer`. However, we lose type inference of kernel here (cause `'a` and `'t` are not related in any way). Therefore, we could pass any value of any type to `SetArguments` and it will not be checked by compiler. Apparently, the translator will still have to be modified.

Reply:
You are right. The translator should be modified in order to support ```Buffer<'t>```, and this type parameter (``` 't ```) should be removed.

Reply:
Fixed.